tests/regression/src/lib.rs

```diff
@@ -1,48 +1,72 @@
-pub fn set_config(
-    cipher_prefs: &security::Policy,
-    keypair: CertKeyPair,
-) -> Result<s2n_tls::config::Config, Error> {
-    let mut builder = Builder::new();
-    builder
-        .set_security_policy(cipher_prefs)
-        .expect("Unable to set config cipher preferences");
-    builder
-        .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
-        .expect("Unable to set a host verify callback.");
-    builder
-        .load_pem(keypair.cert(), keypair.key())
-        .expect("Unable to load cert/pem");
-    builder.trust_pem(keypair.cert()).expect("load cert pem");
-    builder.build()
+pub mod git {
+    use std::process::Command;
+
+    pub fn get_current_commit_hash() -> String {
+        let output = Command::new("git")
+            .args(["rev-parse", "HEAD"])
+            .output()
+            .expect("Failed to get commit hash");
+        assert!(output.status.success());
+
+        String::from_utf8(output.stdout)
+            .expect("Invalid UTF-8 in commit hash")
+            .trim()
+            .to_string()
+    }
+
+    pub fn is_commit_in_log(commit: &str) -> bool {
+        let output = Command::new("git")
+            .args(["log", "--pretty=format:%H"])
+            .output()
+            .expect("Failed to execute git log");
+        assert!(output.status.success());
+
+        let log = String::from_utf8(output.stdout).expect("Invalid UTF-8 in git log output");
+        log.lines().any(|line| line == commit)
+    }
+
+    /// Returns true if `commit1` is older than `commit2`
+    pub fn is_older_commit(commit1: &str, commit2: &str) -> bool {
+        let status = Command::new("git")
+            .args(["merge-base", "--is-ancestor", commit1, commit2])
+            .status()
+            .expect("Failed to execute git merge-base");
+        status.success()
+    }
+
+    pub fn extract_commit_hash(file: &str) -> String {
+        // input: "target/$commit_id/test_name.raw"
+        // output: "$commit_id"
+        file.split("target/")
+            .nth(1)
+            .and_then(|s| s.split('/').next())
+            .map(|s| s.to_string())
+            .unwrap_or_default() // This will return an empty string if the Option is None
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use s2n_tls::{
+        config::Builder,
+        security,
+        testing::{CertKeyPair, InsecureAcceptAllCertificatesHandler},
+    };
+    type Error = s2n_tls::error::Error;
     use super::*;
     use crabgrind as cg;
     use s2n_tls::testing::TestPair;
     use std::{
         env,
-        fs::{create_dir_all, File},
-        io::{self, BufRead, Write},
-        path::Path,
+        fs::{create_dir_all, write},
+        io::{self, BufRead},
         process::Command,
     };
 
-    /// Configurable threshold for regression testing. 
-    /// Tests will fail if the instruction count difference is greater than the value of this constant.
-    const MAX_DIFF: u64 = 1_000_000; 
+    const MAX_DIFF: i64 = 1_000;
```

Comment:
Not blocking this PR but I would recommend that each harness defines this value, especially for ones that don't expect to have any variability.

Reply:
Agreed, James and I discussed this as something to include in the next PR.